### PR TITLE
Fix: Unable to set platform defaults

### PR DESCRIPTION
**Pull request by piotrek6641**
**What?**  
Compare whether LastSpaceID matches SpaceID from MiddlewareManager
**Why?**  
When changing tokens the CLI session was holding old Manager ID which was leading to receiving error 403
https://github.com/scramjetorg/transform-hub/issues/850

**Usage:**

- Have 2 accounts on a platform 
- Change account using generated token on a website
-  Try listing instances or make any other operation.

**Review checks:**

These aspects need to be checked by the reviewer:

- [ ] Verify and confirm operation (please post a screenshot) 
- [ ] All STH tests pass
- [ ] All [Scramjet Cloud Platform](https://docs.scramjet.org/platform) tests pass
- [ ] Documentation is updated or no changes

### CODE REVIEW

File: packages/cli/src/lib/platform/common.ts [hunk 1]
```diff
@@ -36,14 +36,17 @@ export const getMiddlewareClient = (): MiddlewareClient => {
...
 
 export const setPlatformDefaults = async () => {
+    const middlewareClient = getMiddlewareClient();
+    const middlewareManager = await middlewareClient.getManagers();
+
     const { lastSpaceId, lastHubId } = sessionConfig.get();
 
     if (lastSpaceId || lastHubId) {
-        return false;
+        if (middlewareManager[0].id === lastSpaceId) {
```
Comment (patuwwy) [inline packages/cli/src/lib/platform/common.ts:45]:
```suggestion
        if (middlewareManager[0]?.id === lastSpaceId) {
```

**Review by patuwwy — APPROVED** [review]

**Review by alicja-gruzdz — APPROVED** [review]
Verified by switching between two diffrent accounts using the same default profile and executing a couple of listing commands - it worked fine, thanks. Let's merge it and I will release a new STH this week.

File: packages/cli/src/lib/platform/common.ts [hunk 2]
```diff
@@ -36,14 +36,17 @@ export const getMiddlewareClient = (): MiddlewareClient => {
 };
 
 export const setPlatformDefaults = async () => {
+    const middlewareClient = getMiddlewareClient();
+    const middlewareManager = await middlewareClient.getManagers();
```
Comment (patuwwy) [inline packages/cli/src/lib/platform/common.ts:40]: managers

Comment (patuwwy) [reply packages/cli/src/lib/platform/common.ts:40]: and can be reused at line 51 without requesting MW second time

**Review by patuwwy — CHANGES_REQUESTED** [review]
see comments

File: packages/cli/src/lib/platform/common.ts (hunk 2 above)
Comment (piotrek6641) [reply packages/cli/src/lib/platform/common.ts]: fixed with [91f7afc](https://github.com/scramjetorg/transform-hub/pull/883/commits/91f7afc83a4bfcf7ba2aaf7c67c5c047af3abf9d)

**Review by patuwwy — APPROVED** [review]

**Pull request merged**